src/vma/dev/allocator.cpp

```diff
@@ -92,7 +92,8 @@ void* vma_allocator::alloc_and_reg_mr(size_t size, ib_ctx_handler *p_ib_ctx_h)
 		}
 		else {
 			__log_info_dbg("Huge pages allocation passed successfully");
-			if (!register_memory(size, p_ib_ctx_h, m_non_contig_access_mr)) {
+			if (g_p_ib_ctx_handler_collection->get_ib_cxt_list() &&
```

Comment:
i dont agree with the check. register_memory will do nothing if the list is empty
get_ib_cxt_list() should stay as is.

Reply:
why not to move this check to register function?